Cargo.lock

```diff
@@ -3325,13 +3326,13 @@ dependencies = [
 [[package]]
 name = "libp2p"
 version = "0.39.1"
-source = "git+https://github.com/sigp/rust-libp2p?rev=323cae1d08112052740834aa1fb262ae43e6f783#323cae1d08112052740834aa1fb262ae43e6f783"
+source = "git+https://github.com/pawanjay176/rust-libp2p?branch=total-connection-limit#75fd53ec5407a58ae1ff600fd1c68ea49079364a"
```

Comment:
So it seems we need two PRs ahead of upstream libp2p:

* https://github.com/libp2p/rust-libp2p/pull/2137 and
* https://github.com/libp2p/rust-libp2p/pull/2103

Do we have rough timelines for when those PRs might merge, or should we push ahead with a release built on the `sigp` fork of libp2p?

Reply:
My understanding was that `Add a total established connection limit` PR would be merged in the next version (might be wrong here though).

@ethDreamer would have better idea about the `message_id_fn` closure PR.

Anyway, I agree that it would be better to push ahead with building on top of the sigp fork. I'll make a PR

Reply:
Updated dependency in 0656056

Reply:
@michaelsproul @pawanjay176 
The `message_id_fn` closure has been merged into the upstream fork of gossipsub

Reply:
Awesome!

I reckon we go ahead and merge this PR, and then re-target upstream once Age's gossipsub PR is merged (https://github.com/libp2p/rust-libp2p/pull/2175)